src/common/commonutils/DeviceInfoUtils.c

```diff
@@ -32,49 +32,52 @@ void RemovePrefixBlanks(char* target)
 
 void RemovePrefixUpTo(char* target, char marker)
 {
-    if (NULL == target)
+    size_t targetLength = 0;
+    char* equalSign = NULL;
+    
+    if ((NULL == target) || (0 == (targetLength = strnlen(target, MAX_STRING_LENGTH))))
     {
         return;
     }
-
-    int targetLength =(int)strlen(target);
-    char* equalSign = strchr(target, marker);
-
-    if (equalSign)
+    
+    if ((NULL != (equalSign = strchr(target, marker))) && 
+        (0 < (targetLength = strnlen(equalSign + 1, MAX_STRING_LENGTH))))
     {
-        targetLength = strlen(equalSign + 1);
         memmove(target, equalSign + 1, targetLength);
         target[targetLength] = 0;
     }
 }
 
 void RemovePrefixUpToString(char* target, const char* marker)
 {
-    if ((NULL == target) || (NULL == marker))
+    size_t targetLength = 0, markerLength = 0;
+    char* equalSign = NULL;
+    
+    if ((NULL == target) || (0 >= (targetLength = strnlen(target, MAX_STRING_LENGTH))) || 
+        (NULL == marker) || (0 >= (markerLength = strnlen(marker, MAX_STRING_LENGTH))) ||
+        (markerLength >= targetLength))
     {
         return;
     }
 
-    int targetLength = 0;
-    char* equalSign = strstr(target, marker);
-
-    if (equalSign)
+    if (NULL != (equalSign = strstr(target, marker)))
     {
-        targetLength = (int)strlen(equalSign);
+        targetLength = strnlen(equalSign, targetLength + 1);
         memmove(target, equalSign, targetLength);
         target[targetLength] = 0;
     }
 }
 
 void RemoveTrailingBlanks(char* target)
 {
+    int i = 0;
+
     if (NULL == target)
     {
         return;
     }
 
-    int targetLength = (int)strlen(target);
-    int i = targetLength;
+    i = (int)strnlen(target, MAX_STRING_LENGTH);

```

Comment:
Thanks for the feedback! I agree that it would be useful to catch and report strings that are too long. May simply remove this restriction here and allow strings for any length, same as it was before. TBD.

Reply:
Reverted by going back to allow strings of any length to be processed.